File: basic/1004.cpp
#include <algorithm>
#include <cstdio>
const int maxn = 105;

struct Node {
    char name[maxn];
    char id[maxn];
    int score;
    void input() { scanf("%s%s%d", name, id, &score); }
    void print() { printf("%s %s\n", name, id); }
    bool operator<(const Node &rhs) const { return score > rhs.score; }
};
Node student[maxn];

int main() {
    int n;
    scanf("%d", &n);
    for (int i = 0; i < n; ++i)
        student[i].input();
    std::sort(student, student + n);
    student[0].print();
    student[n-1].print();
    return 0;
}

File: basic/1004.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Collections;

class Node implements Comparable<Node> {
    private String name;
    private String id;
    private int score;

    public Node(String _name, String _id, int _score) {
        this.name = _name;
        this.id = _id;
        this.score = _score;
    }

    public void print() {
        System.out.printf("%s %s\n", name, id, score);
    }

    @Override
    public int compareTo(Node rhs) {
        return this.score - rhs.score;
    }
}

class Main {
    static Scanner in;

    public static void main(String[] args) {
        in = new Scanner(System.in);
        int n = in.nextInt();
        ArrayList<Node> students = new ArrayList<Node>();
        for (int i = 0; i < n; ++i) {
            String name = in.next();
            String id = in.next();
            int score = in.nextInt();
            students.add(new Node(name, id, score));
        }
        Collections.sort(students);

        
        students.get(n-1).print();
        students.get(0).print();

        in.close();
    }
}
